### PR TITLE
New test case for rolling update

### PR DESCRIPTION
**Pull request by Katka92**
### What does this PR do?
This PR adds new test class for rolling update. It creates and run workspace, rollout che server, waits until rollout is done and then checks if workspace is still running. 

### What issues does this PR fix or reference?
https://github.com/redhat-developer/rh-che/issues/726
https://github.com/redhat-developer/che-functional-tests/issues/326

### CODE REVIEW

**Comment by riuvshin** [comment]
Can one of the admins verify this patch?

**Comment by Katka92** [comment]
@rhopp could you please review too?

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/rolling/RollingUpdateStrategyWithStartedWorkspaceTest.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+        hotUpdateUtil.executeMasterPodUpdateCommand();
+
+
+        // check that che is updated
+        hotUpdateUtil.waitMasterPodRevision(currentRevision + 1);
+        hotUpdateUtil.waitFullMasterPodUpdate(currentRevision);
+
+        // execute stop-start commands for existing workspaces
+        assertEquals(cheTestSystemClient.getStatus(), SystemStatus.RUNNING);
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/rolling/RollingUpdateStrategyWithStartedWorkspaceTest.java]: It's interesting to know in which way `cheTestSystemClient.getStatus()` executes stop-start commands?

**Comment by dmytro-ndp** [comment]
Looks okay for me do not taking into account redundant empty lines.
@Ohrimenko1988 what do you think about proposed test RollingUpdateStrategyWithStartedWorkspaceTest.java? Doesn't it duplicate other tests of package org.eclipse.che.selenium.hotupdate.rolling which you created previously?

**Comment by Ohrimenko1988** [comment]
@Katka92 please take a look at this two tests:
 - https://github.com/eclipse/che/blob/master/selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/rolling/RollingUpdateStrategyWithWorkspacesStartStopTest.java
 - https://github.com/eclipse/che/blob/master/selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/rolling/RollingUpdateStrategyWithEditorTest.java
It looks like your use case is already covered by this two tests.

**Comment by Ohrimenko1988** [comment]
Yes, you are right. But in my opinion, we should just add this part from your test https://github.com/eclipse/che/blob/7e2b53809b64ee236780afc6d896614b08e456d3/selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/rolling/RollingUpdateStrategyWithStartedWorkspaceTest.java#L57-L59

to the end of this test
https://github.com/eclipse/che/blob/master/selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/rolling/RollingUpdateStrategyWithWorkspacesStartStopTest.java#L68

**Comment by Katka92** [comment]
@Ohrimenko1988 Sorry, I still can't get it. You have one running and one stopped workspace. When rollout is in progress, you change statuses of both of them. So there is none workspace running for all time when rollout is executed - so nothing to check. And I can't simply create third workspace and start it, because on rh-che we can have only one workspace running. Or maybe I didn't get what you meant by the comment.

**Review by Ohrimenko1988 — APPROVED** [review]

**Comment by Ohrimenko1988** [comment]
@Katka92 Sorry for the delay

**Comment by Katka92** [comment]
@Ohrimenko1988 Nothing at all. I'm glad we've discussed it.

**Comment by dmytro-ndp** [comment]
ci-build

**Comment by riuvshin** [comment]
ci-bulid

**Comment by riuvshin** [comment]
ci-build

**Comment by riuvshin** [comment]
build failed, please reformat code to fix.

**Comment by dmytro-ndp** [comment]
@Katka92: well, I ran the tests of package `org.eclipse.che.selenium.hotupdate.rolling` locally and they failed on _Eclipse Che Multiuser_ on _OCP_ infrastructure because of error:
```
[ERROR] startStopWorkspaceFunctionsShouldBeAvailableDuringRollingUpdate(org.eclipse.che.selenium.hotupdate.rolling.RollingUpdateStrategyWithStartedWorkspaceTest)  Time elapsed: 30.249 s  <<< FAILURE!
org.eclipse.che.selenium.core.utils.process.ProcessAgentException: Can't process command '/tmp/oc rollout latest $(oc get dc | grep che | awk '{print $1}')'.
	at org.eclipse.che.selenium.core.utils.process.ProcessAgent.process(ProcessAgent.java:40)
	at org.eclipse.che.selenium.core.executor.OpenShiftCliCommandExecutor.execute(OpenShiftCliCommandExecutor.java:75)
	at org.eclipse.che.selenium.core.executor.hotupdate.HotUpdateUtil.executeMasterPodUpdateCommand(HotUpdateUtil.java:113)
	at org.eclipse.che.selenium.hotupdate.rolling.RollingUpdateStrategyWithStartedWorkspaceTest.startStopWorkspaceFunctionsShouldBeAvailableDuringRollingUpdate(RollingUpdateStrategyWithStartedWorkspaceTest.java:59)
...
Caused by: java.lang.Exception: Output: ; Error: /bin/bash: oc: command not found
error: one deployment config name is needed as argument..
	at org.eclipse.che.selenium.core.utils.process.ProcessAgent.processOutput(ProcessAgent.java:57)
	at org.eclipse.che.selenium.core.utils.process.ProcessAgent.process(ProcessAgent.java:37)
	... 16 more
```

**Comment by Katka92** [comment]
Thank you, I'll investigate it as soon as possible.

**Comment by dmytro-ndp** [comment]
@Katka92 please take into account, that `UPDATE_COMMAND` variable is not completely ready-to-use CLI command - it is the command of OpenShift CLI application, and so you can't just add `$(oc get dc | grep che | awk '{print $1}')'`. Lets compare it with command from error message:
`UPDATE_COMMAND =
      "rollout latest $(oc get dc | grep che | awk '{print $1}')"`
vs
`Can't process command '/tmp/oc rollout latest $(oc get dc | grep che | awk '{print $1}')'`

**Comment by dmytro-ndp** [comment]
@Katka92: please, take fixup on rh-che.

**Comment by dmytro-ndp** [comment]
BTW: locally hot update selenium tests were executed successfully against Eclipse Che Multiuser on OpenShift with rolling update strategy:
```
[INFO] Running TestSuite
2018-08-16 15:13:22,712[main]             [INFO ] [UserCheDefaultTestUserProvider 69]   - User name='admin', id='b07e3a58-ed50-4a6e-be17-fcf49ff8b242' is being used as admin
2018-08-16 15:13:27,763[main]             [INFO ] [.c.s.c.c.k.c.KeycloakCliClient 80]   - Default test user with name='user1534421600889' and id='6e2226a2-2d3d-4af1-8fb2-d92f38401ab4' has been created.
2018-08-16 15:13:29,685[main]             [INFO ] [.e.c.s.c.i.SeleniumTestHandler 207]  - Starting suite 'Surefire suite' with 3 test methods.
2018-08-16 15:13:29,700[pool-1-thread-1]  [INFO ] [.e.c.s.c.i.SeleniumTestHandler 242]  - Dependencies injection in org.eclipse.che.selenium.hotupdate.rolling.RollingUpdateStrategyWithEditorTest
2018-08-16 15:13:30,486[onPool-worker-1]  [INFO ] [.CheTestWorkspaceServiceClient 78]   - Workspace name='workspacejvq4pj' and id='workspace31ajaukgx0l5fia4' created
2018-08-16 15:13:52,140[onPool-worker-1]  [INFO ] [o.e.c.s.c.w.TestWorkspaceImpl 64]    - Workspace name='workspacejvq4pj' id='workspace31ajaukgx0l5fia4' started in 21 sec.
2018-08-16 15:13:53,192[pool-1-thread-1]  [INFO ] [.e.c.s.c.i.SeleniumTestHandler 165]  - Starting test #1 RollingUpdateStrategyWithEditorTest.shouldUpdateMasterByRollingStrategyWithAccessibleEditorInProcess. Passed: 0, failed: 0, skipped: 0.
2018-08-16 15:15:16,667[pool-1-thread-1]  [INFO ] [.e.c.s.c.i.SeleniumTestHandler 165]  - Starting test #2 RollingUpdateStrategyWithStartedWorkspaceTest.startStopWorkspaceFunctionsShouldBeAvailableDuringRollingUpdate. Passed: 1, failed: 0, skipped: 0.
2018-08-16 15:15:16,667[pool-1-thread-1]  [INFO ] [.e.c.s.c.i.SeleniumTestHandler 353]  - Processing @PreDestroy annotation in org.eclipse.che.selenium.hotupdate.rolling.RollingUpdateStrategyWithEditorTest
2018-08-16 15:15:19,080[pool-1-thread-1]  [INFO ] [ractTestWorkspaceServiceClient 120]  - Workspace name='workspacejvq4pj', id='workspace31ajaukgx0l5fia4', username='user1534421600889' removed
2018-08-16 15:15:19,081[pool-1-thread-1]  [INFO ] [.e.c.s.c.i.SeleniumTestHandler 242]  - Dependencies injection in org.eclipse.che.selenium.hotupdate.rolling.RollingUpdateStrategyWithStartedWorkspaceTest
2018-08-16 15:15:19,627[onPool-worker-2]  [INFO ] [.CheTestWorkspaceServiceClient 78]   - Workspace name='workspace4hxwms' and id='workspace0uckm576ezmcn3cq' created
2018-08-16 15:15:38,248[onPool-worker-2]  [INFO ] [o.e.c.s.c.w.TestWorkspaceImpl 64]    - Workspace name='workspace4hxwms' id='workspace0uckm576ezmcn3cq' started in 18 sec.
2018-08-16 15:16:49,231[pool-1-thread-1]  [INFO ] [.e.c.s.c.i.SeleniumTestHandler 165]  - Starting test #3 RollingUpdateStrategyWithWorkspacesStartStopTest.startStopWorkspaceFunctionsShouldBeAvailableDuringRollingUpdate. Passed: 2, failed: 0, skipped: 0.
2018-08-16 15:16:49,231[pool-1-thread-1]  [INFO ] [.e.c.s.c.i.SeleniumTestHandler 353]  - Processing @PreDestroy annotation in org.eclipse.che.selenium.hotupdate.rolling.RollingUpdateStrategyWithStartedWorkspaceTest
2018-08-16 15:16:50,629[pool-1-thread-1]  [INFO ] [ractTestWorkspaceServiceClient 120]  - Workspace name='workspace4hxwms', id='workspace0uckm576ezmcn3cq', username='user1534421600889' removed
2018-08-16 15:16:50,712[pool-1-thread-1]  [INFO ] [.e.c.s.c.i.SeleniumTestHandler 242]  - Dependencies injection in org.eclipse.che.selenium.hotupdate.rolling.RollingUpdateStrategyWithWorkspacesStartStopTest
2018-08-16 15:16:51,091[onPool-worker-3]  [INFO ] [.CheTestWorkspaceServiceClient 78]   - Workspace name='workspace810c2s' and id='workspace0tt5x260h9r87p6t' created
2018-08-16 15:17:12,809[onPool-worker-3]  [INFO ] [o.e.c.s.c.w.TestWorkspaceImpl 64]    - Workspace name='workspace810c2s' id='workspace0tt5x260h9r87p6t' started in 21 sec.
2018-08-16 15:17:13,713[onPool-worker-3]  [INFO ] [.CheTestWorkspaceServiceClient 78]   - Workspace name='workspaceile704' and id='workspacefossdkmvvvu52epi' created
2018-08-16 15:18:46,288[main]             [INFO ] [.e.c.s.c.i.SeleniumTestHandler 526]  - Cleaning up test environment...
2018-08-16 15:18:46,289[main]             [INFO ] [.e.c.s.c.i.SeleniumTestHandler 353]  - Processing @PreDestroy annotation in org.eclipse.che.selenium.hotupdate.rolling.RollingUpdateStrategyWithWorkspacesStartStopTest
2018-08-16 15:18:47,816[main]             [INFO ] [ractTestWorkspaceServiceClient 120]  - Workspace name='workspaceile704', id='workspacefossdkmvvvu52epi', username='user1534421600889' removed
2018-08-16 15:18:48,116[main]             [INFO ] [ractTestWorkspaceServiceClient 120]  - Workspace name='workspace810c2s', id='workspace0tt5x260h9r87p6t', username='user1534421600889' removed
2018-08-16 15:18:49,809[main]             [INFO ] [.c.s.c.c.k.c.KeycloakCliClient 178]  - Test user with name='user1534421600889' has been removed.
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 329.554 s - in TestSuite
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
```

**Comment by Katka92** [comment]
I tried to run test against rh-che localy and it is green. So I think we can merge if you are OK with that. Thank you for your help!

**Pull request merged**